lib/prism/parse_result.rb

```diff
@@ -12,15 +12,13 @@ class Source
     attr_accessor :start_line
 
     # The list of newline byte offsets in the source code.
-    attr_reader :offsets
+    attr_accessor :offsets
```

Comment:
I'd rather not have this be `attr_accessor`, it's not meant to be set later.

Reply:
I changed both start_line and offsets to be `attr_reader`.
And then used `#instance_variable_set/rb_iv_set()` to set them internally as needed.

Reply:
Actually we could even go further and remove the `attr_reader` for `offsets`, like the Java Source object does.
Since offsets is an implementation detail of how Source#line works, and it might change.
But that seems out of scope for this PR.